tensorflow-core/tensorflow-core-native/src/main/java/org/tensorflow/internal/c_api/presets/tensorflow.java

```diff
@@ -58,31 +58,12 @@
             "tensorflow/c/eager/c_api.h",
             "tensorflow/c/eager/c_api_experimental.h",
 
-// ---------------------------------------------------------
-// NOTICE CUSTOM GRADIENT: In TF Java 0.6.0, custom gradient registration has been disabled due to the precarity of the
-// Java bindings issued from the internal TensorFlow C++ APIs using JavaCPP. These APIs are subject to changes between
-// TF releases, which make them difficult to maintain. If you want to reenable this feature, please uncomment the code
-// between all occurrences of this notice and the "END OF CUSTOM GRADIENT" mention.
-// ---------------------------------------------------------
-//            "absl/status/status.h",
-//            "tensorflow/tsl/platform/status.h",
-//            "tensorflow/tsl/platform/default/status.h",
-//            "tensorflow/core/util/port.h",
-//            "tensorflow/c/tf_status_helper.h",
-//            "tensorflow/c/kernels.h",
-//            "tensorflow/c/ops.h",
-//            "tensorflow/c/c_api_internal.h",
-//            "tensorflow/c/eager/abstract_context.h",
-//            "tensorflow/c/eager/abstract_tensor_handle.h",
-//            "tensorflow/c/eager/gradients.h",
-//            "tensorflow/cc/framework/ops.h",
-//            "tensorflow/cc/framework/scope.h",
-//            "tensorflow/cc/framework/grad_op_registry.h",
-//            "tensorflow/core/graph/graph.h",
-//            "tensorflow_adapters.h", // Keep this at the bottom, since it is not a real tensorflow header but one that we maintain (see main resources)
-// ---------------------------------------------------------
-// END OF CUSTOM GRADIENT
-// ---------------------------------------------------------
+            // Following are C API extensions maintained within TF Java, see src/main/native.
+            // Binding directly the C++ API with JavaCPP turned out to be too precarious between different releases,
+            // so it is simpler to write our own C API only exposing what we need from it.
+            "graph.h",
+            "scope.h",
+            "gradients.h",
```

Comment:
Should we name these `tfj_gradients.h` etc?

Reply:
I'm questioning this too... `tf_` or `tfj_`.. I kind of like that our extensions _blends_ with the original C API, that is why I left it the class name with the `TF_` prefix, though it makes it more explicit it is custom to TF Java using a different prefix (though do users care?)

Reply:
I guess it doesn't matter too much. They'd likely get new names if we tried to contribute it back to the TF C API anyway.